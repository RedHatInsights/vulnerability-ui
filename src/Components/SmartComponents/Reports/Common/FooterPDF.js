import React from 'react';
import propTypes from 'prop-types';
import { View, Text } from '@react-pdf/renderer';
import styles from './styles';
import messages from '../../../../Messages';
import { intlFormatWithBold } from '../ReportsHelper';

const FooterPDF = ({ hasRule, hasExploit }) => {

    return (
        <View>
            { hasExploit && (
                <Text style={styles.footer}>
                    {intlFormatWithBold(messages.knownExploitPDFFooter)}
                </Text>
            )}

            { hasRule && (
                <Text style={styles.footer}>
                    {intlFormatWithBold(messages.securityRulePDFFooter)}
                </Text>
            )}

        </View>
    );
};

FooterPDF.propTypes = {
    hasRule: propTypes.bool,
    hasExploit: propTypes.bool
};

export default FooterPDF;
