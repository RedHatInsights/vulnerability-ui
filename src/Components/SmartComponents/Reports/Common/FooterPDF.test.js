import FooterPDF from './FooterPDF';
import toJson from 'enzyme-to-json';

// trick to fix formatWithBold function work
jest.mock('./../ReportsHelper', () => ({ ...jest.requireActual('./../ReportsHelper'), intlFormatWithBold: (msg) => msg.defaultMessage }));

describe('FooterPDF component', () => {
    it('Should match snapshot when it\'s empty.', () => {
        const wrapper = mount(
            <FooterPDF />
        );        
          
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should match snapshot when it has exploit and security rule.', () => {
        const wrapper = mount(
            <FooterPDF hasRule hasExploit />
        );        
          
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
