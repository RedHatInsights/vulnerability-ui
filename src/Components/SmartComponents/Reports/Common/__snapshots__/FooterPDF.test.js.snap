// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterPDF component Should match snapshot when it has exploit and security rule. 1`] = `
<FooterPDF
  hasExploit={true}
  hasRule={true}
>
  <VIEW>
    <TEXT
      style={
        {
          "color": "#6A6E73",
          "fontSize": 8,
        }
      }
    >
      &lt;b&gt;Known exploit:&lt;/b&gt; This CVE is identified with a “Known exploit” label because Red Hat has determined this CVE has a public exploit. This CVE is unpatched on your system. CVEs with this label should be addressed with high priority due to the risks posed by them. “Known exploit” does not mean we have taken steps to determine if the CVE has been exploited in your environment.
    </TEXT>
    <TEXT
      style={
        {
          "color": "#6A6E73",
          "fontSize": 8,
        }
      }
    >
      &lt;b&gt;Security rule:&lt;/b&gt; Indicates a security rule associated with this CVE. Security rules are written by Red Hat to help you configure your systems.
    </TEXT>
  </VIEW>
</FooterPDF>
`;

exports[`FooterPDF component Should match snapshot when it's empty. 1`] = `
<FooterPDF>
  <VIEW />
</FooterPDF>
`;
